src/BuiltInTools/dotnet-watch/CommandLine/CommandLineOptions.cs

```diff
@@ -69,12 +75,13 @@ internal sealed class CommandLineOptions
             noHotReloadOption,
             NonInteractiveOption
         ];
-
-        // Options we need to know about that are passed through to the subcommand:
-        var shortProjectOption = new Option<string>("-p") { Hidden = true, Arity = ArgumentArity.ZeroOrOne, AllowMultipleArgumentsPerToken = false };
-        var longProjectOption = new Option<string>("--project") { Hidden = true, Arity = ArgumentArity.ZeroOrOne, AllowMultipleArgumentsPerToken = false };
-        var launchProfileOption = new Option<string>("--launch-profile", "-lp") { Hidden = true, Arity = ArgumentArity.ZeroOrOne, AllowMultipleArgumentsPerToken = false };
-        var noLaunchProfileOption = new Option<bool>("--no-launch-profile") { Hidden = true, Arity = ArgumentArity.Zero };
+        Option[] forwardToRunOptions =
```

Comment:
Instead of hard-coding this list, can we check `!command.Options.Contains(optionResult.Option)` in `GetCommandArguments`?